File: solutions/python/18.py
def to_weird_case(string):
    result = ''
    index = 0
    for char in string:
        if index % 2:
            result += char.lower()
        else:
            result += char.upper()
        if char.isalpha():
            index += 1
        else:
            index = 0
    return result

File: solutions/c/18.c
#include <stddef.h>

struct IntegerPair {
  int key;
  int value;
};

int gcd(int n, int m) {
  while (m > 0) {
    int temp = m;
    m = n % m;
    n = temp;
  }
  return n;
}

int lcm(int n, int m) {
  if (m == 0) {
    return 0;
  }
  return (n * m) / gcd(n, m);
}

int sum_differences_between_products_and_LCMs(struct IntegerPair *pairs, size_t count) {
  int result = 0;
  for (size_t index = 0; index < count; ++index) {
    int product = pairs[index].key * pairs[index].value;
    int last_common_multiple = lcm(pairs[index].key, pairs[index].value);
    result += product - last_common_multiple;
  }
  return result;
}